pythontest.py

```diff
@@ -42,71 +42,88 @@ def reading_file(file):
-
-headerCheck(origin)
-headerCheck(destination)
-
-#def file_copy(origin,destination):
-    # Creating first match criteria via ID
-    # Read all IDs of origin file and converting into regular array
-    #origin_ids = origin['id'].to_numpy()
-
-    # Read all IDs of destination file and converting into regular array
-    #destination_ids = destination['id']
-    # Looping through all origin ids and check if it exists in destination
-    # for id_value in destination_ids:
-    #     # If we find the id in the destination file we copy over the value
-    #     if id_value in origin_ids:
-    #         # Get the value of the origin id for the price
-    #         price = origin['price'][origin.id == id_value]
-    #         # Set value of destination file price to this price
-    #         destination['price'][destination.id == id_value] = price
-
+            sys.exit('The file does not have the right header columns. Please provide this column: ' + column)
 
 
 def file_copy(origin, destination):
     # iterate over ids in the destination file
+    # Set-up counter for matches
+    i = 0
+    # Set-up counter for where the price was already correct
+    z = 0
     for index in range(len(destination)):
         # find all items in origin that match that item id
         filterId = origin['id'] == destination.loc[index, 'id']
         matchesId = origin.loc[filterId]
+        # create counter for changed id lines
+
         # if there are no matched we do nothing, otherwise we copy over the price
         if len(matchesId) > 0:
             destination.loc[index, 'price'] = matchesId.iloc[0]['price']
-        # find all items in origin that match the name column
-        # TODO: Convert all names to lower or upper case
-        filterName = origin['name'] == destination.loc[index, 'name']
+            i += 1
+        else:
+            # find all items in origin that match the name column
+            filterName = origin['name'].str.upper() == destination.loc[index, 'name'].upper()
+
+            matchesName = origin.loc[filterName]
+            if len(matchesName) > 0:
+                if destination.loc[index, 'price'] == matchesName.iloc[0]['price']:
+                    z += 1
+                else:
+                    destination.loc[index, 'price'] = matchesName.iloc[0]['price']
+                    i += 1
+
+    # Report for amount of rows in origin
+    lines_origin = len(origin.axes[0])
+    print('Number of items in the origin file ', lines_origin)
+
+    # Report for amount of rows in destination
+    lines_destination = len(destination.axes[0])
+    print('Number of items in the destination file ', lines_destination)
+
+    # Report for changed lines
+    print('We have changed ', i, ' Items')
+    # Report for items that could not be matched
+    print('We were not able to match ', (lines_origin - ( i + z )) ,' item(s)')
```

Comment:
Here the ones that couldn't be matched refers to the prices that we want to align. So the total lines in the destination file instead of the origin (but the requirement was ambiguous and the developer is always right)